src/NinjaTools.FluentMockServer.API/Infrastructure/SetupRepository.cs

```diff
@@ -1,54 +1,41 @@
+using System;
 using System.Collections.Generic;
-using System.Linq;
-using System.Net;
 using JetBrains.Annotations;
 using Microsoft.AspNetCore.Http;
 using NinjaTools.FluentMockServer.API.Models;
-using HttpResponse = NinjaTools.FluentMockServer.API.Models.HttpResponse;
+using NinjaTools.FluentMockServer.API.Services;
 
 namespace NinjaTools.FluentMockServer.API.Infrastructure
 {
     internal sealed class SetupRepository : ISetupRepository
     {
         private readonly List<Setup> _setups;
 
-        public SetupRepository()
+        private readonly ILogService _logService;
+
+        public SetupRepository(ILogService logService)
         {
-#if DEBUG
-            _setups = new List<Setup>()
-            {
-                new Setup
-                {
-                    Matcher = new RequestMatcher
-                    {
-                        Path = "/some/test"
-                    },
-                    Action = new ResponseAction
-                    {
-                      Response = new HttpResponse()
-                      {
-                          StatusCode = (int) HttpStatusCode.Accepted,
-                          Body = "hello world!"
-                      }
-                    }
-                }
-            };
-#else
+            _logService = logService;
             _setups = new List<Setup>();
-#endif
         }
 
         public IEnumerable<Setup> GetAll() => _setups;
 
-        public void Add(Setup setup) => _setups.Add(setup);
+        public void Add(Setup setup)
+        {
+            _setups.Add(setup);
+            _logService.Log(log => log.SetupCreated(setup));
+        }
 
         /// <inheritdoc />
         [CanBeNull]
         public Setup? TryGetMatchingSetup([NotNull] HttpContext context)
         {
-            return GetAll().FirstOrDefault(s => s.Matcher.IsMatch(context)) is {} setup
-                ? setup
-                : null;
+            throw new  NotImplementedException();
+            // return GetAll().FirstOrDefault(s => s.Matcher?.IsMatch(context) ?? false) is {} setup
+            //     ? setup
+            //     : null;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4802644)